pkg/workload/tpcc/tpcc.go

```diff
@@ -1096,7 +1101,22 @@ func (w *tpcc) executeTx(
 	}
 
 	if w.txnRetries {
-		return onTxnStartDuration, crdbpgx.ExecuteTx(ctx, conn, txOpts, txnFuncWithStartFuncs)
+		for i := 0; i <= maxRetries; i++ {
```

Comment:
Sorry, I should have thought of this earlier... what about exposing the number of attempted retries as a metric, to help troubleshooting/diagnosing? On the other hand, I'm not aware of a specific metric for savepoints, which would have conveyed the same information.

Reply:
Not sure about metrics, but I added a log line similar to [this](https://github.com/cockroachdb/cockroach/blob/c83c57d354741ac36740894f7387c014eb6c09fe/pkg/kv/txn.go#L1133). We should be able to see in the logs if a transaction is stuck retrying.